src/metabase/api/collection.clj

```diff
@@ -64,7 +111,11 @@
                          :id
                          (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
              :order-by [[:%lower.name :asc]]}) collections
-      ;; include Root Collection at beginning or results if archived isn't `true`
+          ;; Remove other users' personal collections
+          (if exclude-other-user-collections?
+            (remove-other-users-personal-collections collections)
```

Comment:
your overload reaches out to `api/*current-user-id*` in an optional overload. I think you should just pass it in. Makes it obvious when testing to supply it. Otherwise it will get a null user-id if people don't realize it grabs that itself.